### DISCUSSION TITLE
TypeScript Can't Find Prisma Client Model Types

### DISCUSSION THREAD

**Discussion by gangradem** [prisma/prisma#17742] [Q&A]
I have a model called User.

When I generate the Prisma client, I can see that there are lots of types generated for that model automatically. I can see them in the client file.

But Typescript can only find one of the types: UserScalarFieldEnum.

UserSelect returns undefined.

All other such types I can see in the client return undefined.

I'm on Windows 11. I've tried regenerating the client several times. I've even tried recreating the entire thing from scratch.

Even when I followed a Typescript + Prisma setup tutorial from the Prisma website, I still see this problem.

What am I doing wrong here? I have wasted hours on this problem now.

I am sure it's something simple that I'm missing. But I just can't figure it out.

**Comment by gangradem** [comment]
I got it now....

I don't have the ability to delete this thread. Can someone do it for me? :)

**Reply by arosemena** [reply]
It may be worth posting your solution since someone having your same problem may arrive at this thread

**Reply by nurul3101** [reply]
Yes, adding the solution that worked for you should be helpful to others :)

**Comment by gangradem** [comment]
The error was my mistake. I was trying to console log the types. That's why it was returning undefined.

I'm new to TypeScript. (if it's not apparent from this 🤣 )

**Reply by nurul3101** [reply]
Thanks for letting us know! :)

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that haven’t had any recent activity and appear to be outdated.

If this discussion is still important to you or unresolved, we’d love to hear from you! Feel free to reopen it or start a new one with updated details.

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and being part of the community!
